contract/contracts/NewInterfaces.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.6.0;
+
+contract DistributerInterface {
+    // TODO Add events
+
+    string name;
+    address tokenHolder;
+    mapping(address => address) tokenRefundDestinationList;
+
+    function balanceOf(address token) virtual external view returns (uint256) {}
+
+    function addToken() virtual external {}
+
+    function createCampaign(
+        address token,
+        string memory campaignInfoCid,
+        string memory recipientsCid,
+        string memory recipientsNum,
+        uint256 startDate
+    ) virtual external {}
+
+    // Future functionality
+    // function updateTokenHolder(address newTokenHolder) external; // onlyOwner
+}
+
+contract CampaignInterface {
+    // TODO Add events
+
+    string campaignInfoCid; // Contains campaign name and description as JSON
+    string recipientsCid; // Contains recipients value as JSON
+    uint32 recipientsNum;
+    uint32 claimedNum;
+    address refundDestination; // Use when campaign is cancelled
+
+    // Check msg.sender is claimable
+    function isClaimable(address token) virtual external view returns (bool) {}
+
+    // Claim tokens
+    function claim(address token) virtual external {}
+
+    function cancelCampaign() virtual external {}
+
```

Comment:
I think we need a function to transfer tokens when the campaign ends.

Reply:
I created `cancelCampaign()` for the purpose of that. It cancels campaign and refund tokens.

Reply:
I understand `Cancel` means stopping the campaign before its start.

Campaign `End` means that a situation where fan cannot claim a token after the campaign's end time. And after that state, I think we need a function that only executes the refund of the token.

I think it's possible to achieve both with a single function, but I think the roles are different.

What do you think?

Reply:
I'm agree with you. I will add `endCampaign()` and add some description to both.